systers_portal/meetup/views.py

```diff
@@ -801,79 +774,50 @@ def post(self, request):
-#                     # if user filters by distance
-#                     geolocator = Nominatim(timeout=6)
-#                     user_loc = (geolocator.geocode(location))
-#                     user_point = Point((float)(user_loc.raw['lon']), (float)(user_loc.raw['lat']))
-#                     meetup_loc = (geolocator.geocode(meetup.meetup_location.location))
-#                     meetup_point = Point((float)(meetup_loc.raw['lon']),
-#                                          (float)(meetup_loc.raw['lat']))
-#                     distance = (int)(user_point.distance(meetup_point)) * 100
-#                     results.sort(key=operator.itemgetter('distance'))
-#                     unit = 'kilometers from your location'
-#
-#                 results.append({'date': meetup.date,
-#                                 'meetup': meetup.title,
-#                                 'location': meetup.meetup_location.name,
-#                                 'location_slug': meetup.meetup_location.slug,
-#                                 'meetup_slug': meetup.slug,
-#                                 'distance': distance,
-#                                 'unit': unit})
-#
-#             results.sort(key=operator.itemgetter('date'))
-#             return JsonResponse({'search_results': results}, safe=False)
+
+class UpcomingMeetupsSearchView(ListView):
+    """Search Upcoming Meetups By  Keyword and Filter Date and Distance"""
+    template_name = "meetup/list_meetup.html"
+    model = Meetup
+
+    @csrf_exempt
+    def post(self, request):
+        if request.method == 'POST':
+            keyword = request.POST.get('keyword')
+            location = request.POST.get('location')
+            searched_meetups = Meetup.objects.filter(Q(date__gte=datetime.date.today()),
+                                                     Q(title__icontains=keyword))
+            results = list()
+            unit = ''
+            for meetup in searched_meetups:
+                distance = ''
+                geolocator = Nominatim(timeout=6)
+                g = GeoIP2()
+                if location == "Current Location":
+                    client_ip, is_routable = get_client_ip(request)
+                    if is_routable:
+                        lat, long = g.lat_lon(client_ip)
+                    else:
+                            lat, long = g.lat_lon("google.com")
+                    user_point = Point((float)(long),
+                                       (float)(lat))
+                else:
+                    user_loc = (geolocator.geocode(location))
+                    user_point = Point((float)(user_loc.raw['lon']), (float)(user_loc.raw['lat']))
+                meetup_loc = (geolocator.geocode(meetup.meetup_location))
+                meetup_point = Point((float)(meetup_loc.raw['lon']),
+                                     (float)(meetup_loc.raw['lat']))
+                distance = (int)(user_point.distance(meetup_point)) * 100
```

Comment:
extra brackets, please note that these are just functions, what you are doing is basically evaluating an expression `(int)` and then calling on it

Reply:
Sure, will take care of it